src/codegen/riscv64/macro-assembler-riscv64.cc

```diff
@@ -1466,7 +1466,15 @@ void TurboAssembler::li(Register rd, Operand j, LiFlags mode) {
   DCHECK(!j.is_reg());
   BlockTrampolinePoolScope block_trampoline_pool(this);
   if (!MustUseReg(j.rmode()) && mode == OPTIMIZE_SIZE) {
-    RV_li(rd, j.immediate());
+    UseScratchRegisterScope temps(this);
+    int count = li_count(j.immediate(), temps.hasAvailable());
+    int neg_count = li_count(-j.immediate(), temps.hasAvailable());
+    if ((count - neg_count) > 1) {
+      RV_li(rd, ~j.immediate());
+      not_(rd, rd);
```

Comment:
In the call to `li_count` you use the negative operator, but in the call to `RV_li`, you use the bitwise not `~`, followed by a `not_`. Shouldn't those match?